src/vcpkg/base/files.cpp

```diff
@@ -1200,7 +1170,24 @@ namespace vcpkg
             }
 #endif // ^^^ !defined(_WIN32)
         }
-        virtual bool remove(const path& target, std::error_code& ec) override { return stdfs::remove(target, ec); }
+        virtual bool remove(const path& target, std::error_code& ec) override
+        {
+            bool result = stdfs::remove(target, ec);
+#if defined(_WIN32)
+            if (ec && ec == std::error_code(ERROR_ACCESS_DENIED, std::system_category()))
+            {
+                set_writeable(target, ec);
+                if (ec)
+                {
+                    return false;
+                }
+
+                stdfs::remove(target, ec);
```

Comment:
If this succeeds, the function will still return false -- is that intended?
```suggestion
                return stdfs::remove(target, ec);
```